stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/channel/ChannelController.kt

```diff
@@ -682,13 +688,28 @@ public class ChannelController internal constructor(
             }.toMutableList()
         }.also {
             message.attachments = it
-
             // RepositoryFacade::insertMessage is implemented as upsert, therefore we need to delete the message first
             domainImpl.repos.deleteChannelMessage(message)
             domainImpl.repos.insertMessage(message)
         }
     }
 
+    private fun updateAttachmentUploadState(messageId: String, uploadId: String, newState: Attachment.UploadState) {
+        val message = _messages.value[messageId]
+        if (message != null) {
+            val newAttachments = message.attachments.fold(emptyList<Attachment>()) { acc, item ->
+                if (item.uploadId == uploadId) {
+                    acc + item.copy(uploadState = newState)
+                } else {
+                    acc + item
+                }
+            }
```

Comment:
This creates a new list for each iteration, so going through 100 messages creates 100 new list instances. 

Wouldn't a `map` operation work to achieve what we need here, modifying a single item, and keeping all others? That creates just a single new list and pre-allocates it to the appropriate size as well.

Reply:
Map operation won't work here because the state flow won't emit the new value as it will already be modified. That's why the creation of a new list is necessary unless we modify our StateFlow for such cases.